android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -812,6 +900,26 @@ public UrlDeviceBuilder setDescription(String description) {
     }
 
     /**
+     * Set wifi-direct MAC address.
+     * @param MAC address corresonding to UrlDevice.
+     * @return The builder with address
+     */
+    public UrlDeviceBuilder setWifiAddress(String address) {
+      addExtra("direct", address);
```

Comment:
Use something more explicit than "direct".  Also, I'd prefer if these were made static members instead of literal strings.